src/components/gamespec/Board.js

```diff
@@ -11,15 +11,50 @@ const boxTarget = {
       let offset = monitor.getClientOffset(),
           item = monitor.getItem();
 
+      let pieces = Object.assign([], props.getPieces());
       let items = props.getItems();
+      let copy = Object.assign([], items);
       let rect = component.refs.stage.getStage().getContainer().getBoundingClientRect();
       offset.x = offset.x - rect.left;
       offset.y = offset.y - rect.top;
       let image = item.image;
-      items.push({image, offset});
-      props.setItems(items);
+      let indice = [];
```

Comment:
A lot of logic in this drag function and the function below is getting reused, extract it out in a single function

Reply:
but I need to return two values if I extract the function which is not a good idea for the javascript

Reply:
I think I will need to return the copy and pieces within one function

Reply:
Upto you, I haven't read the code thoroughly, but I can see same (a little bit different) logic in both of them. So they can be shifted to one function.

Reply:
Yep, we found this out, but we also noticed that if we want to write them into a new function, we need to return two values, which is not recommended on google